src/modules/SelectBox/index.js

```diff
@@ -22,11 +22,28 @@ export default class SelectBox extends Component {
 
   state = {
     open: false,
+    providedValues: false,
     selected: ''
   }
 
+  componentDidMount() {
+    this._checkTextValue();
+  }
+
+
   // PRIVATE
 
+  _checkTextValue = () => {
+    const { list } = this.props;
+    if (!!list.length) {
```

Comment:
We can make this cleaner without the heavy conditional nesting. I would rather see
```
if (!list.length || (typeof list[0] !== 'object')) {
  return;
}

...
```